service/controller/resource/app/resource.go

```diff
@@ -13,24 +14,29 @@ const (
 
 // Config represents the configuration used to create a new chartconfig service.
 type Config struct {
-	G8sClient versioned.Interface
-	K8sClient kubernetes.Interface
-	Logger    micrologger.Logger
+	ClusterClient *clusterclient.Client
```

Comment:
As mentioned before if we extend the `operatorversions` resource we don't need the cluster client here.